nameserver/mocks_test.go

```diff
@@ -75,6 +75,10 @@ func (mz *mockedZoneWithRecords) AddRecord(ident string, name string, ip net.IP)
 	notImplWarn()
 	return nil
 }
+func (mz *mockedZoneWithRecords) DeleteRecords(ident *string, ip *net.IP, name *string) int {
```

Comment:
Or perhaps those other methods should go away entirely, since their functionality is subsumed by `DeleteRecords`? And if you don't like points, how about `DeleteRecords` takes a predicate function?